controlplane/test/override.test.ts

```diff
@@ -0,0 +1,221 @@
+import { EnumStatusCode } from '@wundergraph/cosmo-connect/dist/common/common_pb';
+import { joinLabel } from '@wundergraph/cosmo-shared';
+import { afterAll, beforeAll, describe, expect, test } from 'vitest';
+import { SchemaCheckRepository } from '../src/core/repositories/SchemaCheckRepository.js';
+import { InspectorOperationResult } from '../src/core/services/SchemaUsageTrafficInspector.js';
+import { afterAllSetup, beforeAllSetup, genID, genUniqueLabel } from '../src/core/test-util.js';
+import { SetupTest } from './test-util.js';
+
+let dbname = '';
+
+describe('Overrides', (ctx) => {
+  beforeAll(async () => {
+    dbname = await beforeAllSetup();
+  });
+
+  afterAll(async () => {
+    await afterAllSetup(dbname);
+  });
+
+  test('Should be able to detect overrides', async (testContext) => {
+    const { client, server } = await SetupTest(testContext, dbname);
+
+    const fedGraphName = genID('fedGraph');
+    const label = genUniqueLabel();
+
+    const createFedGraphRes = await client.createFederatedGraph({
+      name: fedGraphName,
+      namespace: 'default',
+      routingUrl: 'http://localhost:8081',
+      labelMatchers: [joinLabel(label)],
+    });
+
+    expect(createFedGraphRes.response?.code).toBe(EnumStatusCode.OK);
+
+    const graphRes = await client.getFederatedGraphByName({
+      name: fedGraphName,
+      namespace: 'default',
+    });
+
+    const namespacesRes = await client.getNamespaces({});
+    const namespace = namespacesRes.namespaces.find((n) => n.name === graphRes.graph?.namespace);
+
+    expect(namespace).toBeDefined();
+    expect(graphRes.response?.code).toBe(EnumStatusCode.OK);
+
+    const createOverrideRes = await client.createOperationOverrides({
+      graphName: graphRes.graph?.name,
+      namespace: graphRes.graph?.namespace,
+      operationHash: 'hash1',
+      operationName: 'op1',
+      changes: [
+        {
+          changeType: 'FIELD_TYPE_CHANGED',
+          path: 'A.field',
+        },
+      ],
+    });
+    expect(createOverrideRes.response?.code).toBe(EnumStatusCode.OK);
+
+    const schemaCheckRepo = new SchemaCheckRepository(server.db);
```

Comment:
This will test the repository not the client methods. We can try to add a full integration tests by passing a spied `chClient` to `setuptests`

Reply:
Done